src/notification.c

```diff
@@ -280,55 +283,96 @@ char *notification_extract_markup_urls(char **str_ptr) {
 }
 
 /*
- * Create notification struct and initialise everything to NULL,
- * this function is guaranteed to return a valid pointer.
+ * Create notification struct and initialise all fields with either
+ *  - the default (if it's not needed to be freed later)
+ *  - its undefined representation (NULL, -1)
+ *
+ * This function is guaranteed to return a valid pointer.
+ * @Returns: The generated notification
  */
 notification *notification_create(void)
 {
-        return g_malloc0(sizeof(notification));
-}
+        //FIXME: handle OOM
```

Comment:
From [glibs docs](https://developer.gnome.org/glib/stable/glib-Memory-Allocation.html)
> If any call to allocate memory fails, the application is terminated. This also means that there is no need to check if the call succeeded.

Means we don't need to handle OOM.

Reply:
Thanks. This wasn't clear for me. Only [g_try_malloc0](https://developer.gnome.org/glib/stable/glib-Memory-Allocation.html#g-try-malloc0) mentions the behavior of `g_malloc0`.